etna/pipeline/hierarchical_pipeline.py

```diff
@@ -312,25 +312,25 @@ def _forecast_prediction_interval(
         self, ts: TSDataset, predictions: TSDataset, quantiles: Sequence[float], n_folds: int
     ) -> TSDataset:
         """Add prediction intervals to the forecasts."""
-        # TODO: fix this: what if during backtest KeyboardInterrupt is raised
         self.forecast, self.raw_forecast = self.raw_forecast, self.forecast  # type: ignore
 
-        if self.ts is None:
-            raise ValueError("Pipeline is not fitted! Fit the Pipeline before calling forecast method.")
-
-        # TODO: rework intervals estimation for `BottomUpReconciliator`
+        try:
+            if self.ts is None:
+                raise ValueError("Pipeline is not fitted! Fit the Pipeline before calling forecast method.")
```

Comment:
Can we move this check before the initial swap to make the code a bit more readable?